src/components/DnsSettingsForm.js

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) [2020] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState } from 'react';
+import cockpit from 'cockpit';
+import {
+    Alert,
+    FormGroup,
+    TextInput
+} from '@patternfly/react-core';
+import { useNetworkDispatch, updateDnsSettings } from '../context/network';
+import ModalForm from './ModalForm';
+import IPInput from './IPInput';
+import { isValidIP, isValidDomain } from '../lib/utils';
+import { createDnsSettings } from '../lib/model/dnsSettings';
+import deep_equal from 'deep-equal';
+
+const _ = cockpit.gettext;
+
+const sanitizeList = (value, delimiter = " ") => {
+    return value.split(delimiter).filter(Boolean)
+            .join(delimiter);
+};
+
+const DnsSettingsForm = ({ isOpen, onClose, dns }) => {
+    const dispatch = useNetworkDispatch();
+    const [nameserver1, setNameserver1] = useState(dns.nameServers[0]);
+    const [nameserver2, setNameserver2] = useState(dns.nameServers[1]);
+    const [nameserver3, setNameserver3] = useState(dns.nameServers[2]);
+    const [policy, setPolicy] = useState(dns.policy);
+    const [searchListInput, setSearchListInput] = useState(dns.searchList.join(" "));
+    const [errorMessages, setErrorMessages] = useState([]);
+
+    const validate = () => {
+        const errors = [];
+
+        // Clean previous error messages
+        setErrorMessages([]);
+
+        if (searchList().some((d) => !isValidDomain(d))) {
+            errors.push({
+                key: 'invalid-searchlist',
+                message: _("There are invalid searchList domains")
```

Comment:
I would prefer "There are invalid domains in the search list" or "The search list contains invalid domains" or, at least, separate "searchList".